src/database/models/entitytype.js

```diff
@@ -9,21 +9,47 @@ module.exports = (sequelize, DataTypes) => {
 				primaryKey: true,
 				type: DataTypes.INTEGER,
 			},
-			value: { type: DataTypes.STRING, allowNull: false },
+			value: { type: DataTypes.STRING, allowNull: false, unique: true },
 			label: { type: DataTypes.STRING, allowNull: false },
-			status: { type: DataTypes.STRING, allowNull: false },
-			type: { type: DataTypes.STRING, allowNull: false },
+			status: { type: DataTypes.STRING, allowNull: false, defaultValue: 'ACTIVE' },
 			created_by: { type: DataTypes.INTEGER, allowNull: false },
 			updated_by: { type: DataTypes.INTEGER, allowNull: false },
 			allow_filtering: {
 				type: DataTypes.BOOLEAN,
 				allowNull: false,
 			},
+			data_type: { type: DataTypes.STRING, allowNull: false },
 		},
 		{ sequelize, modelName: 'EntityType', tableName: 'entity_types', freezeTableName: true, paranoid: true }
 	)
 	EntityType.associate = (models) => {
-		EntityType.hasMany(models.Entity, { foreignKey: 'entity_type_id' })
+		EntityType.hasMany(models.Entity, {
+			foreignKey: 'entity_type_id',
+			as: 'entities',
+			scope: {
+				deleted_at: null, // Only associate with active EntityType records
+			},
+		})
 	}
+
+	EntityType.addHook('beforeDestroy', async (instance, options) => {
+		try {
+			// Soft-delete only the associated Entity records with matching entity_type_id
+			await sequelize.models.Entity.update(
+				{ deleted_at: new Date() }, // Set the deleted_at column to the current timestamp
```

Comment:
Are we clear on all ambiguity regarding these deleteAt workflows?

Reply:
[Clarified]: We are not using sequelize built in functions to delete, since all of these are soft-deletes and sequelize doesn't really perform as expected with soft-deletes.